OBAKitCore/Network/Operations/NetworkOperation.swift

```diff
@@ -20,6 +20,9 @@ public enum APIError: Error, LocalizedError {
     case noResponseBody
     case requestFailure(HTTPURLResponse)
 
+    /// A `404` error.
+    case requestNotFound(HTTPURLResponse)
```

Comment:
I can't think of a reason why we'd differentiate a `real 404` and an `effective 404`, so I'm treating it as equals here. This removes the  `statusCodeIsEffectively404` flag.

Reply:
You might be able to find the reason I added this if you look at the git blame for this commit. There's some OBA REST API that returns a nonsensical 200 response to some request that should really be a 404. Perhaps the assumptions in the rest of the system have changed such that a misleading 200 response will no longer cause issues, but that needs to be run down before this gets removed.

Reply:
Oops, I wasn't clear. I didn't remove the handling of empty 200. See https://github.com/OneBusAway/onebusaway-ios/pull/622/files#r1060340430